Ports from Strawberry Jam into Communal Helper

catapillie opened the pull request:
We're been asked to port [Strawberry Jam](https://gamebanana.com/mods/424541) entities into Communal Helper for maintenance and potential updates, bugfixes, etc...

Notes and important stuff:
* Entities ported from SJ must have their name changed. This is the pattern used: `CommunalHelper/SJ/...`.
* Graphics will be put under `Graphics/Atlases/Gameplay/objects/CommunalHelper/strawberryJam/`.
* Sprite data (`.xml`) can be pasted `Graphics/CommunalHelper/Sprites.xml` for now. (Should it be changed to a separate .xml?)
* Lönn plugins and Ahorn plugins will be put under `Loenn/entities/strawberry_jam` and `Ahorn/entities/strawberryJam` respectively.
* Mark Strawberry Jam en_gb.lang entries with `[Strawberry Jam]` for clarity. For Lönn plugins, we'll append `[Strawberry Jam]` in placement names, and in ahorn plugins, we'll write `(Strawberry Jam) (Communal Helper)`.

TODO:

* **Air Time Music Controller**
    - [x] Port implementation from `AirTimeMusicController.cs`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
* **Bullet Time Controller**
    - [x] Port implementation from `BulletTimeController.cs`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
* **Expiring Dash Refill**
    - [x] Port implementation from `ExpiringDashRefill.cs`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
* ~~**Dash Count Trigger**~~ _(cancelled port)_
* **Show Hitbox Trigger**
    - [x] Port implementation from `ShowHitboxTrigger.cs`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
* ~~**Switch Crate**~~ _(moved to [Batteries](https://github.com/CommunalHelper/Batteries/issues/3))_
* ~~**Switch Crate Holder**~~ _(moved to [Batteries](https://github.com/CommunalHelper/Batteries/issues/3))_
* ~~**Switch Door**~~ _(moved to [Batteries](https://github.com/CommunalHelper/Batteries/issues/3))_
* ~~**UFO**~~ _(planned for another update, plugins staying hidden for now)_
* **Grab Temple Gate**
    - [x] Port implementation from `GrabTempleGate.cs`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
* **Loop Block**
    - [x] Port implementation from `LoopBlock.cs`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
    - [x] Add missing SFX
* **Solar Elevator**
    - [x] Port implementation from `SolarElevator.cs`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
    - [x] Change default dialog and sfx entries (in entity option)
    - [x] Add missing SFXs
* ~~**Dash Zip Mover**~~ _(may be ported another way, likely as an entity option to already an existing entity in Communal Helper)_
* **Photosensitive Flag Controller**
    - [x] Port implementation from `PhotosensitiveFlagController.cs`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
* **Exploding Strawberry**
    - [x] Port implementation from `ExplodingStrawberry.cs` and `CustomEntitySprites.xml`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
* **Wormhole Booster**
    - [x] Port implementation from `WormholeBooster.cs` and `CustomEntitySprites.xml`
    - [x] Write Lönn plugin
    - [x] Copy Ahorn plugin
    
Other things:
* [x] Copy SJ's SFX bank events into CommunalHelper's (renaming some events)

l-Luna committed:
bullet time controller entity

l-Luna committed:
working bullet time controller + loenn plugin

catapillie committed:
ported loop block plugin to lönn

catapillie committed:
lönn lang file tweak for sj ports

catapillie committed:
ported loop block impl

l-Luna committed:
move bullet time controller under `SJ2021`

l-Luna committed:
re-rename BTC

l-Luna committed:
rename loop block (under SJ/)

catapillie committed:
add grab temple gate lönn plugin

catapillie committed:
import grab temple gate graphics and sprite data

l-Luna committed:
`ShowHitboxTrigger`

catapillie committed:
add grab temple gate implementation

l-Luna commented:
minor `ShowHitboxTrigger` bugfix: it no longer stops sound sources from being displayed while the player has the debug menu open inside a show hitbox trigger

l-Luna committed:
Show Hitbox Trigger ahorn plugin

catapillie committed:
improved grab temple gate plugin rendering

catapillie committed:
fix grab temple gate freeze-frame bug, (not merged in sj yet)

catapillie committed:
add grab temple gate lang entries

catapillie committed:
Merge branch 'sj-ports' of https://github.com/CommunalHelper/CommunalHelper into sj-ports

catapillie committed:
fix wrong name in lang file

l-Luna committed:
bullet time controller ahorn plugin

l-Luna committed:
Merge remote-tracking branch 'origin/sj-ports' into sj-ports

l-Luna committed:
fix ahorn plugin name

l-Luna committed:
change BTC loenn plugin defaults for parity

catapillie committed:
copy grab temple gate, loop block, solar elevator ahorn plugins

catapillie committed:
Merge branch 'sj-ports' of https://github.com/CommunalHelper/CommunalHelper into sj-ports

l-Luna committed:
Photosensitive Flag Controller // move BTC icon to proper place

l-Luna committed:
Merge remote-tracking branch 'origin/sj-ports' into sj-ports

l-Luna committed:
Photosensitive Flag Controller ahorn plugin

catapillie committed:
import solar elevator graphics

catapillie committed:
edit solar elevator ahorn plugin with new texture paths

catapillie committed:
port solar elevator plugin to lönn

catapillie committed:
Merge branch 'sj-ports' of https://github.com/CommunalHelper/CommunalHelper into sj-ports

catapillie committed:
fix typo in solar elevator lönn plugin

catapillie committed:
add solar elevator implementation

catapillie committed:
add default solar elevator holdable hint dialog

catapillie committed:
added solar elevator lang file entries

l-Luna commented:
Switch Door relies on batteries' Battery Gate, and Switch Crate stuff should probably be implemented as an improvement to Batteries instead.

l-Luna committed:
exploding strawberry

catapillie committed:
partially port SJ SFX bank

copied over sfx that are likely to be used in the SJ-CH port, and discarded other sfx that are used in single maps

catapillie committed:
changed solar elevator dialog hint constant

catapillie committed:
changed solar elevator default sfx entries

catapillie committed:
add back loopblock sfx

catapillie committed:
add air time music controller

catapillie committed:
add expiring dash refill

catapillie commented:
> Switch Door relies on batteries' Battery Gate, and Switch Crate stuff should probably be implemented as an improvement to Batteries instead.

Moving those to [Batteries](https://github.com/CommunalHelper/Batteries) since they are under the Communal Helper organization. Ticking them off in this PR.

catapillie commented:
Also going to skip porting Dash Zip Movers. They can be ported as a new (better) mechanic or as an option to Connected Zip Movers.

catapillie committed:
add dash count trigger

catapillie committed:
move showhitboxtrigger ahorn plugin

catapillie committed:
add missing tooltips in ahorn lang file

l-Luna committed:
wormhole booster

l-Luna committed:
fix wormhole booster placement description

l-Luna committed:
basic UFO impl

l-Luna committed:
add ufo sprites

l-Luna committed:
work on UFO beam

catapillie committed:
update loop block texture (recent sj change)

catapillie commented on Ahorn/entities/strawberryJam/UFO.jl:
```diff
@@ -0,0 +1,38 @@
+﻿module SJ2021UFO
```
```suggestion
CommunalHelperUFO
```

catapillie committed:
Change UFO ahorn plugin module name

catapillie commented:
Going to hide the UFO plugins as they're being worked on for a future update.

catapillie committed:
hide UFO plugins

catapillie commented:
Cancelling the Dash Count Trigger port (Kahuna says it's not a good idea, because bad/cursed implementation or whatever)

catapillie committed:
remove Dash Count Trigger port
